src/os_posix/os_union_fs.c

```diff
@@ -536,168 +540,139 @@ __union_fs_file_lock(WT_FILE_HANDLE *file_handle, WT_SESSION *wt_session, bool l
     return (fh->destination.fh->fh_lock(fh->destination.fh, wt_session, lock));
 }
 
-// This may be easier with a partial_start, partial_end, partial macro.
-/*
- * Given a read or a write must fall either in an existing extend or on the edge of it the two
- * choices are NONE or FULL.
- *
- * This only works by assuming that the block manager is the only thing that will use this and that
- * it only reads and writes full blocks. If that changes this code will unceremoniously fall over.
- */
-typedef enum { NONE, FULL } RW_SERVICE_LEVEL;
-
 /*
- * __dest_can_service_rw --
- *     Return if a the read or write can be serviced by the destination.
+ * __union_remove_extlist_hole --
+ *     Track that we wrote something by removing its hole from the extent list.
  */
-static RW_SERVICE_LEVEL
-__dest_can_service_rw(
+static int
+__union_remove_extlist_hole(
   WT_UNION_FILE_HANDLE *union_fh, WT_SESSION_IMPL *session, wt_off_t offset, size_t len)
 {
-    WT_UNION_ALLOC_LIST *alloc;
-    wt_off_t rw_end;
+    WT_UNION_HOLE_LIST *hole, *tmp, *new, *prev_hole;
+    wt_off_t write_end;
 
-    WT_UNUSED(session);
+    __wt_verbose_debug2(session, WT_VERB_FILEOPS, "REMOVE HOLE %s: %ld-%ld", union_fh->iface.name,
+      offset, offset + (wt_off_t)len - 1);
 
-    /* Walk the extent list until extent offset + size > read offset + size. */
-    rw_end = offset + (wt_off_t)len;
+    write_end = offset + (wt_off_t)len - 1;
```

Comment:
Can we define an OFF_END() macro which is called by EXTENT_END so this logic isn't duplicated?

Reply:
Done in [6bda525](https://github.com/wiredtiger/wiredtiger/pull/11268/commits/6bda52511bdfb6846178368721fa51f1c212d0d0)